_posts/2017-01-09-Jak-psat-php-Zavislosti+verzovaci-systemy+debugging.md

```diff
@@ -0,0 +1,38 @@
+---
+layout: post
+title: Jak psát PHP? (2/3) - Závislosti + verzovací systémy + debugging
+---
+
+# Manažer pro závislosti - Composer
+[Composer](https://getcomposer.org/) je software řešící závislosti v kódu. V podstatě se stará o to, aby všechny balíčky cizího kódu (na kterých závisíte, proto závislosti) byly stažené, ve správné verzi a nebily se mezi sebou. Composer je umí sám stahovat, komtrolovat nové verze i bezpečně odstraňovat. V PHP dlouho něco takového chybělo a všichni jsme rádi, že tu je (kdo z vás vzpomíná na stahování knihoven z php.net při rozcházení cizího projektu?). 
+
+Pro základní používání Composeru si vystačíte si příkazy pro přidání, aktualizaci a odebrání balíčků a nemusíte ho dál řešit. Ovládá se pomocí příkazového řádku a souboru `/composer.json`. Ten obsahuje seznam všech balíčků a jejich dalších závislosti. Samotné soubory s kódem jsou potom standardně uloženy ve složce `/vendor`. 
+
+Jakmile začnete používat framework nebo cizí kód, Composer se stane neocenitelným společníkem. Rozhodně se vyplatí ho začít používat co nejdříve. Za zmínku stojí, že pomocí něj můžou ostatní používat jednoduše kód váš. Více o jeho instalaci a použití najdete na <https://getcomposer.org/doc/00-intro.md>. 
+
+Obzvláště v již existujícím kódu můžete narazit na starší systém pro správu PHP rozšíření [PEAR](https://pear.php.net/). Ten se používá pro správu PHP komponent, neřeší ovšem závislosti tak elegantně jako Composer a neposkytuje žádnou jinou zásadnější výhodu. Některé knihovny jsou ale historicky dostupné jen přes PEAR, v některých aplikacích se proto používá dodnes. 
+
+# Verzovací systém (Version Control System, zkráceně verzování nebo VCS)
+Jsou mezi námi dlouho a používá je míň lidí, než by bylo vhodné. Verzovací systémy slouží ke snadné správě kódu v čase. Můžete se podívat, kdy jaká změna v kódu nastala a který vývojář ji provedl. Umožňují jednoduše provést "rollback" (návrat v čase) nebo souběžný vývoj více různých větví programu a jejich následné sloučení do jednoho finálního projektu. Při vyvíjení v týmu větším než tři lidi už jsou prakticky nepostradatelné. 
```

Comment:
Asi bych přidal ukázka, odkazy na Github projekty. Klidně menší.

Aj na linux-kernel, to mě třeba příjemně překvapilo, že i něco je open-source na Githubu